File: 378. Kth Smallest Element in a Sorted Matrix/Java/Kth Smallest Element in a Sorted Matrix.java
public class Solution {
    public int kthSmallest(int[][] matrix, int k) {
        class Item {
            int value;
            int i;
            int j;

            public Item(int value, int i, int j) {
                this.value = value;
                this.i = i;
                this.j = j;
            }
        }
        int m = matrix.length;
        int n = matrix[0].length;
        PriorityQueue<Item> minHeap = new PriorityQueue<>(new Comparator<Item>() {
            @Override
            public int compare(Item o1, Item o2) {
                return o1.value - o2.value;
            }
        });
        for (int j = 0; j < n; j++) {
            minHeap.add(new Item(matrix[0][j], 0, j));
        }
        while (!minHeap.isEmpty()) {
            Item item = minHeap.poll();
            if (item.i < m - 1) {
                minHeap.add(new Item(matrix[item.i + 1][item.j], item.i + 1, item.j));
            }
            k--;
            if (k == 0) {
                return item.value;
            }
        }
        return 0;
    }
}

File: 378. Kth Smallest Element in a Sorted Matrix/Python/Kth Smallest Element in a Sorted Matrix.py
class Solution(object):
    def kthSmallest(self, matrix, k):
        """
        :type matrix: List[List[int]]
        :type k: int
        :rtype: int
        """
        m = len(matrix)
        n = len(matrix[0])
        min_heap = []
        for j in range(n):
            heapq.heappush(min_heap, (matrix[0][j], 0, j))
        while min_heap:
            current, i, j = heapq.heappop(min_heap)
            if i < m - 1:
                heapq.heappush(min_heap, (matrix[i + 1][j], i + 1, j))
            k -= 1
            if k == 0:
                return current
